server/sonar-server/src/main/java/org/sonar/server/component/ws/ComponentDtoToWsComponent.java

```diff
@@ -57,6 +57,9 @@ private ComponentDtoToWsComponent() {
     setNullable(emptyToNull(dto.language()), wsComponent::setLanguage);
     setTags(dto, wsComponent);
     lastAnalysis.ifPresent(analysis -> wsComponent.setAnalysisDate(formatDateTime(analysis.getCreatedAt())));
+    if (Qualifiers.PROJECT.equals(dto.qualifier())) {
+      wsComponent.setPrivate(dto.isPrivate());
```

Comment:
should be set for views also
it's currently always false, but views are a root component and have a private flag